docs/source/howto/codes.rst

```diff
@@ -4,41 +4,264 @@
    | Standardization is good but don't artificially limit the power of a code you are wrapping - or your users will get frustrated.
    | If the code can do it, there should be *some* way to do it with your plugin.
-
  * | **Don't rely on AiiDA internals.**
    | AiiDA's :ref:`public python API<python_api_public_list>` includes anything that you can import via ``from aiida.module import thing``.
    | Functionality at deeper nesting levels is not considered part of the public API and may change between minor AiiDA releases, breaking your plugin.
 
+.. _how-to:codes:parsing:
+
+Parsing the outputs
+===================
+
+The parsing step occurs after the calculation has finished running and all the relevant outputs have been retrieved.
+It is an optional step that allows you to extract relevant information from the output files and store it into AiiDA nodes in formats that are easier and quicker to query and analyze.
+The way to implement this is by extending the base |Parser| class as you can see below:
+
+.. code-block:: python
+   :linenos:
+
+    from aiida.parsers.parser import Parser
+    from aiida import orm
+
+    class ArithmeticAddParser(Parser):
+
+        def parse(self, **kwargs):
+
+            output_folder = self.retrieved
+            output_filename = self.node.get_attribute('output_filename')
+
+            with output_folder.open(output_filename, 'r') as handle:
+                result = int(handle.read())
+
+            self.out('sum', orm.Int(result))
+
+The first command in the example (line 8) shows the way to get the ``retrieve`` folder that was generated by the associated ``CalcJob`` and contains the files included in the ``retrieve_list`` during the ``prepare_for_submission``.
+The second one (line 9) is also accessing a parameter of the |CalcJob|: in this case, the name of the output.
+After unloading this information into local variables, these are then used to open said output file located in the obtained retrieved folder and read the single integer value that was written there by the addition ``code`` (lines 11 and 12).
+
+Finally, it uses the ``out`` method to provide the ``sum`` output of the calculation its final value (of type AiiDA integer, as was specified in the ``define`` section of the associated |CalcJob|).
+It is important to note here that there is no return statement: the output is provided to the ``self.out`` method instead (any returned value is interpreted as an error signal) as an unstored node, and the engine will be in charge of performing the storing process.
+
+In order to use this ``ArithmeticAddParser`` inside an appropriate |CalcJob| (such as the one described in the :ref:`previous section <how-to:codes:interfacing>`), one needs to add it as a `metadata.options.parser_name` input.
+You can leave it for the user to do so when instantiating the desired calculation plugin, or, if you are developing the calculation plugin, you can add it as the default option in the ``define`` method:
```

Comment:
This how-to is written for users, so this sentence is confusing.

Reply:
Changed it, clearer?

In order to use this ``ArithmeticAddParser`` inside an appropriate |CalcJob| (such as the one described in the :ref:`previous section <how-to:codes:interfacing>`), one needs to add it as a `metadata.options.parser_name` input.
You can set a parser as the default option in the |define| method, but note that this choice can be overriden when instantiating the |CalcJob|.

Reply:
Better!